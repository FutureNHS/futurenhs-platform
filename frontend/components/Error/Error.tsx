import React from "react";

import { H2 } from "../H2";
import { MainHeading } from "../MainHeading";

interface ErrorProps {
  title: string;
  description: string;
}

const Error: React.FC<ErrorProps> = ({ title, description }) => {
  return (
    <>
      <MainHeading>Error</MainHeading>
      <H2 title={title} />
      <br />
      <p>{description}</p>
    </>
  );
};

export default Error;
