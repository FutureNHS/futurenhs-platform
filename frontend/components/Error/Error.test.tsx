import React from "react";

import { render } from "@testing-library/react";

import Error from "./Error";

describe(Error, () => {
  it("renders matching snapshot", () => {
    const props = {
      title: "Insufficient Permissions",
      description: "Please contact your administrator",
    };
    const container = render(<Error {...props} />);

    expect(container.asFragment()).toMatchSnapshot();
  });
});
