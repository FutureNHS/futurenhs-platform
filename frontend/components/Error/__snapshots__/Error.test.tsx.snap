// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders matching snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-size: 40px;
  line-height: 1.25;
  overflow-wrap: break-word;
  padding-top: 24px;
  width: 0px;
}

<div
    class="c0"
  >
    <h1
      aria-live="polite"
      class="c1"
    >
      Error
    </h1>
  </div>
  <h2>
    Insufficient Permissions
  </h2>
  <br />
  <p>
    Please contact your administrator
  </p>
</DocumentFragment>
`;
